File: buildSrc/src/main/java/wolkenschloss/pool/BuildDataSourceImage.java
package wolkenschloss.pool;

import org.gradle.api.file.RegularFileProperty;
import org.gradle.api.tasks.CacheableTask;
import org.gradle.api.tasks.Exec;
import org.gradle.api.tasks.InputFile;
import org.gradle.api.tasks.OutputFile;
import org.gradle.api.tasks.PathSensitive;
import org.gradle.api.tasks.PathSensitivity;
import org.gradle.api.tasks.TaskAction;

@CacheableTask
abstract public class BuildDataSourceImage extends Exec {

    public BuildDataSourceImage() {
        commandLine("cloud-localds");
    }

    @InputFile
    @PathSensitive(PathSensitivity.RELATIVE)
    abstract public RegularFileProperty getNetworkConfig();

    @InputFile
    @PathSensitive(PathSensitivity.RELATIVE)
    abstract public RegularFileProperty getUserData();

    @OutputFile
    abstract public RegularFileProperty getDataSourceImage();

    @TaskAction
    @Override
    public void exec() {
        args("--network-config",
                getNetworkConfig().get(),
                getDataSourceImage().get(),
                getUserData().get());

        super.exec();
    }
}

File: buildSrc/src/main/kotlin/wolkenschloss/gradle/testbed/pool/BuildDataSourceImage.kt
package wolkenschloss.gradle.testbed.pool

import org.gradle.api.file.RegularFileProperty
import org.gradle.api.provider.Property
import org.gradle.api.tasks.*
import java.io.File

@CacheableTask
abstract class BuildDataSourceImage : Exec() {
    init {
        commandLine("cloud-localds")
    }

    @get:PathSensitive(PathSensitivity.RELATIVE)
    @get:InputFile
    abstract val networkConfig: Property<File>

    @get:PathSensitive(PathSensitivity.RELATIVE)
    @get:InputFile
    abstract val userData: Property<File>

    @get:OutputFile
    abstract val dataSourceImage: RegularFileProperty

    @TaskAction
    public override fun exec() {
        args(
            "--network-config",
            networkConfig.get(),
            dataSourceImage.get(),
            userData.get()
        )

        super.exec()
    }
}